server/src/internalClusterTest/java/org/opensearch/index/ShardIndexingPressureSettingsIT.java

```diff
@@ -0,0 +1,827 @@
+        if (randomBoolean) {
+            successFuture = client(coordinatingOnlyNode).bulk(bulkRequest);
+            secondSuccessFuture = client(coordinatingOnlyNode).bulk(bulkRequest);
+        } else {
+            successFuture = client(primaryName).bulk(bulkRequest);
+            secondSuccessFuture = client(primaryName).bulk(bulkRequest);
+        }
+
+        // Delay to breach the success time stamp threshold
+        Thread.sleep(25);
+
+        // This request breaches the threshold and hence will be rejected
+        if (randomBoolean) {
+            ShardIndexingPressureTracker coordinatingShardTracker = internalCluster().getInstance(IndexingPressureService.class,
+                coordinatingOnlyNode).getShardIndexingPressure().getShardIndexingPressureTracker(shardId);
+            expectThrows(OpenSearchRejectedExecutionException.class, () -> client(coordinatingOnlyNode).bulk(bulkRequest).actionGet());
+            assertEquals(1, coordinatingShardTracker.getCoordinatingOperationTracker().getRejectionTracker().getTotalRejections());
+            assertEquals(1, coordinatingShardTracker.getCoordinatingOperationTracker().getRejectionTracker()
+                .getLastSuccessfulRequestLimitsBreachedRejections());
+        } else {
+            ShardIndexingPressureTracker primaryShardTracker = internalCluster().getInstance(IndexingPressureService.class, primaryName)
+                .getShardIndexingPressure().getShardIndexingPressureTracker(shardId);
+            expectThrows(OpenSearchRejectedExecutionException.class, () -> client(primaryName).bulk(bulkRequest).actionGet());
+            assertEquals(1, primaryShardTracker.getCoordinatingOperationTracker().getRejectionTracker().getTotalRejections());
+            assertEquals(1, primaryShardTracker.getCoordinatingOperationTracker().getRejectionTracker()
+                .getLastSuccessfulRequestLimitsBreachedRejections());
+        }
+
+        primaryReleasable.close();
+        successFuture.actionGet();
+        secondSuccessFuture.actionGet();
+
+        // Update the outstanding threshold setting to see no rejections
+        ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
+        updateSettingsRequest.persistentSettings(Settings.builder()
+            .put(ShardIndexingPressureMemoryManager.MAX_OUTSTANDING_REQUESTS.getKey(), 10));
+        assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
+
+        // Verify no rejection with similar request pattern
+        primaryReleasable = blockPrimary(primaryName);
+        if (randomBoolean) {
+            successFuture = client(coordinatingOnlyNode).bulk(bulkRequest);
+            secondSuccessFuture = client(coordinatingOnlyNode).bulk(bulkRequest);
+        } else {
+            successFuture = client(primaryName).bulk(bulkRequest);
+            secondSuccessFuture = client(primaryName).bulk(bulkRequest);
+        }
+        Thread.sleep(25);
```

Comment:
These sleeps might cause flaky tests

Reply:
This is to introduce delay between two sequential calls (nothing concurrent), and any delay above ``10ms`` defined as part of ``SUCCESSFUL_REQUEST_ELAPSED_TIMEOUT`` above should be sufficient. 
Since we are not intending on any background activity/tasks to finish, this will not result into flaky tests.